### PR TITLE
Add support for PCA10031 nrf51 wireless dongle

### PR DESCRIPTION
**Pull request by deadprogram**
This PR adds support for the PCA10031 nrf51 wireless dongle.

https://infocenter.nordicsemi.com/pdf/nRF51_Dongle_UG_v1.0.pdf

### CODE REVIEW

**Comment by deadprogram** [comment]
I have tested with the blinky1, blinky2, and echo examples, and all worked as expected.

File: src/machine/board_pca10031.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+// UART pins
+const (
+	UART_TX_PIN = 9
+	UART_RX_PIN = 11
+)
+
+// I2C pins (disabled)
+const (
+	SDA_PIN = 0xff
+	SCL_PIN = 0xff
```
Comment (aykevl) [inline src/machine/board_pca10031.go]: You can set these to `0xffffffff`, which means 'not connected'. I'm not sure what happens with `0xff` but `0xffffffff` (-1) has a defined meaning.

File: src/machine/board_pca10031.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+const HasLowFrequencyCrystal = true
+
+// LED on the pca10031
+const (
+	LED       = LED_RED
+	LED1      = LED_RED
+	LED2      = LED_GREEN
+	LED_RED   = 21
+	LED_GREEN = 22
+	LED_BLUE  = 23
```
Comment (aykevl) [inline src/machine/board_pca10031.go]: You could add a `LED3 = LED_BLUE` here, for completeness.

File: src/machine/board_pca10031.go (hunk 1 above)
Comment (deadprogram) [reply src/machine/board_pca10031.go]: Done.

File: src/machine/board_pca10031.go (hunk 2 above)
Comment (deadprogram) [reply src/machine/board_pca10031.go]: Done.

**Comment by aykevl** [comment]
Sorry I totally forgot, can you also add an entry to the documentation? `docs/targets.rst`

**Comment by aykevl** [comment]
And to `.travis.yml`, for example building examples/blinky1. It makes sure at least building something for the board doesn't get broken in the future.

**Comment by deadprogram** [comment]
Added to docs and to CI build... let's see if it passes.

**Comment by aykevl** [comment]
I only realized these two places that need changes because I was finishing the digispark addition.
And I see .travis.yml has conflicts. I can resolve them if needed during the merge because I caused them seconds ago.

**Comment by deadprogram** [comment]
OK please do resolve as needed.

**Comment by aykevl** [comment]
> /home/travis/gopath/src/github.com/aykevl/tinygo/src/machine/machine_nrf.go:141:16: SDA_PIN (untyped int constant 4294967295) overflows uint8
> /home/travis/gopath/src/github.com/aykevl/tinygo/src/machine/machine_nrf.go:142:16: SCL_PIN (untyped int constant 4294967295) overflows uint8

Right. It may be better to use `int8` for pins, then. For now I think it's best to just keep the pins at 0xff, there is a good chance that will also be interpreted as "not connected" by the hardware.

**Comment by deadprogram** [comment]
Do you want to make that change or shall I?

**Comment by deadprogram** [comment]
I made the change to the pins.

**Comment by deadprogram** [comment]
I resolved the merge conflict in the Travis build and all tests are now passing: https://travis-ci.com/aykevl/tinygo/jobs/159975601#L1361

Seems ready to merge now, unless you have other comments.

**Comment by aykevl** [comment]
Merged in ce0ad1daa281769992207aea440f7f676ba4ce08.

**Pull request closed without merging**